vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -72,6 +83,60 @@
     private Registration validationRegistration;
     private Registration dataProviderListenerRegistration;
 
+    /**
+     * {@link DataController} implementation responsible for data supply from
+     * tied data provider to {@link DataView} and handling {@link SizeChangeEvent}.
+     */
+    protected class CheckboxGroupDataController implements DataController<T> {
```

Comment:
Most of the methods here could be pulled to an `AbstractDataController` or such but it is probably fine for now, we just want to avoid code duplication later on.

Reply:
Let's keep it as it is now. This would be changed probably within #8344